airflow/providers/databricks/plugins/databricks_workflow.py

```diff
@@ -0,0 +1,472 @@
+
+        return ",".join(get_databricks_task_ids(task_group.group_id, tasks_to_run, log))
+
+    @staticmethod
+    def _get_failed_and_skipped_tasks(dr: DagRun) -> list[str]:
+        """
+        Return a list of task IDs for tasks that have failed or have been skipped in the given DagRun.
+
+        :param dr: The DagRun object for which to retrieve failed and skipped tasks.
+
+        :return: A list of task IDs for tasks that have failed or have been skipped.
+        """
+        return [
+            t.task_id
+            for t in dr.get_task_instances(
+                state=[
+                    TaskInstanceState.FAILED,
+                    TaskInstanceState.SKIPPED,
+                    TaskInstanceState.UP_FOR_RETRY,
+                    TaskInstanceState.UPSTREAM_FAILED,
+                    None,
+                ],
+            )
+        ]
+
+
+class WorkflowJobRepairSingleTaskLink(BaseOperatorLink, LoggingMixin):
+    """Construct a link to send a repair request for a single databricks task."""
+
+    name = "Repair a single task"
+
+    def get_link(
+        self,
+        operator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        if not ti_key:
+            ti = get_task_instance(operator, dttm)
+            ti_key = ti.key
+
+        task_group = operator.task_group
+        if not task_group:
+            raise AirflowException("Task group is required for generating repair link.")
+
+        self.log.info(
+            "Creating link to repair a single task for databricks job run %s task %s",
+            task_group.group_id,
+            ti_key.task_id,
+        )
+        dag = airflow_app.dag_bag.get_dag(ti_key.dag_id)
+        task = dag.get_task(ti_key.task_id)
+
+        if ".launch" not in ti_key.task_id:
+            launch_task_id = get_launch_task_id(task_group)
+            ti_key = _get_launch_task_key(ti_key, task_id=launch_task_id)
+        metadata = get_xcom_result(ti_key, "return_value")
+
+        query_params = {
+            "dag_id": ti_key.dag_id,
+            "databricks_conn_id": metadata.conn_id,
+            "databricks_run_id": metadata.run_id,
+            "run_id": ti_key.run_id,
+            "tasks_to_repair": _get_databricks_task_id(task),
+        }
+        return url_for("RepairDatabricksTasks.repair", **query_params)
+
+
+class RepairDatabricksTasks(AirflowBaseView, LoggingMixin):
+    """Repair databricks tasks from Airflow."""
+
+    default_view = "repair"
+
+    @expose("/repair_databricks_job/<string:dag_id>/<string:run_id>/", methods=("GET",))
+    @auth.has_access_dag("POST", DagAccessEntity.RUN)
+    def repair(self, dag_id: str, run_id: str):
+        if not dag_id:
+            return_url = request.referrer or url_for("Airflow.index")
+            flash("No dag_id provided. Cannot repair tasks.")
+            return redirect(return_url)
+
```

Comment:
This check is not needed - if you declare the `dag_id`in the URL pattern you can not enter without.
```suggestion
```

Reply:
I previously had dag_id and run_id in the URL route, but with that unfortunately, some auth manager tests are unhappy.

https://github.com/apache/airflow/actions/runs/9890176859/job/27318136222?pr=40724#step:11:3706

Would you have some recommendations on how to resolve those?

Reply:
Interesting. I have no direct idea but the stack trace tells it in inside the construction of the HTML navigation menu... which should not have any link added from the view as you made the catrgory=None.

Looking at https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html#example and the "nomenu" example, can you try declaring "just" the `"view"` attribute and leave the name+category completely off?

Reply:
thanks a lot @jscheffl . That seems to help :)